framework/src/interfaces/MeshMetaDataInterface.C

```diff
@@ -34,3 +34,42 @@ MeshMetaDataInterface::hasMeshProperty(const std::string & data_name,
   std::string full_name = std::string(SYSTEM) + "/" + prefix + "/" + data_name;
   return _meta_data_app.hasRestartableMetaData(full_name, MooseApp::MESH_META_DATA);
 }
+
+void
+MeshMetaDataInterface::checkMeshMetadataForwardingSetting(
+    const MooseObject * moose_object,
+    const bool has_input,
+    const bool retain_all_input_mesh_metadata,
+    const std::vector<std::string> selected_mesh_metadata_to_retain)
+{
+  if (!has_input)
+  {
+    if (retain_all_input_mesh_metadata)
+      moose_object->paramError("retain_all_input_mesh_metadata",
+                               "In the absence of an input mesh, this parameter must not be true.");
+    if (!selected_mesh_metadata_to_retain.empty())
+      moose_object->paramError("selected_mesh_metadata_to_retain",
+                               "In the absence of an input mesh, this parameter must be empty.");
+  }
+  else if (retain_all_input_mesh_metadata && !selected_mesh_metadata_to_retain.empty())
+    moose_object->paramError(
+        "selected_mesh_metadata_to_retain",
+        "This parameter should not be provided if retain_all_input_mesh_metadata is set true.");
+}
+
+std::vector<std::string>
+MeshMetaDataInterface::identifyMeshMetaData(const MeshGeneratorName prefix) const
+{
+  std::vector<std::string> res;
+  const std::string full_prefix = std::string(SYSTEM) + "/" + prefix + "/";
+  const RestartableDataMap & meta_data =
+      _meta_data_app.getRestartableDataMap(MooseApp::MESH_META_DATA);
+  for (const auto & meta_data_item : meta_data)
+  {
+    if (meta_data_item.first.size() > full_prefix.size())
+      if (meta_data_item.first.compare(0, full_prefix.size(), full_prefix) == 0)
+        res.push_back(meta_data_item.first.substr(
+            full_prefix.size(), meta_data_item.first.size() - full_prefix.size()));
+  }
+  return res;
```

Comment:
so as was discussed with Cody on Shikhar s PR, this MeshMetaDataInterface is only for consumers of metadata, not producers.

so let s moved the code related to forwarding back to MG.C

Reply:
maybe identify is more of a consumer thing though
what do you think

Reply:
I agree with that that `identifyMeshMetaData` can stay in the interface.